src/shared/utils/flattenChildren.js

```diff
@@ -29,8 +30,9 @@ function flattenSingleChildIntoContext(traverseContext, child, name) {
       keyUnique,
       'flattenChildren(...): Encountered two children with the same key, ' +
       '`%s`. Child keys must be unique; when two children share a key, only ' +
-      'the first child will be used.',
-      KeyEscapeUtils.unescape(name)
+      'the first child will be used.%s',
+      KeyEscapeUtils.unescape(name),
+      ReactComponentTreeDevtool.getCurrentStackAddendum(result[name])
```

Comment:
Likewise, you'll want to thread through a debug ID for the parent component here. It looks like there are several places that use flattenChildren so let's add a test for each one to make sure that the stack makes it through correctly.